instrumentation/akka-http-10.0/javaagent/src/test/groovy/AkkaHttpClientInstrumentationTest.groovy

```diff
@@ -25,21 +28,32 @@ class AkkaHttpClientInstrumentationTest extends HttpClientTest implements AgentT
   ActorMaterializer materializer = ActorMaterializer.create(system)
 
   @Override
-  int doRequest(String method, URI uri, Map<String, String> headers, Closure callback) {
+  int doRequest(String method, URI uri, Map<String, String> headers) {
+    return sendRequest(method, uri, headers).toCompletableFuture().get().status().intValue()
+  }
+
+  @Override
+  void doRequestAsync(String method, URI uri, Map<String, String> headers = [:], Consumer<Integer> callback) {
+    sendRequest(method, uri, headers).thenAccept {
+      callback.accept(it.status().intValue())
+    }
+  }
+
+  private CompletionStage<HttpResponse> sendRequest(String method, URI uri, Map<String, String> headers) {
     def request = HttpRequest.create(uri.toString())
       .withMethod(HttpMethods.lookup(method).get())
       .addHeaders(headers.collect { RawHeader.create(it.key, it.value) })
 
-    def response = Http.get(system)
+    return Http.get(system)
       .singleRequest(request, materializer)
-    //.whenComplete { result, error ->
-    // FIXME: Callback should be here instead.
-    //  callback?.call()
-    //}
-      .toCompletableFuture()
-      .get()
-    callback?.call()
-    return response.status().intValue()
+  }
+
+  // TODO(anuraaga): Context leak seems to prevent us from running two asynchronous tests in a row.
```

Comment:
😱 🙀

Reply:
By the way, the reason I know context leak is related is thanks to @jkwatson's awesome logging, showing a client span being suppressed.